### PR TITLE
Remove all the non-tested rootfs

### PR DESCRIPTION
**Pull request by fidencio**
As the centos, fedora, gentoos, and suse rootfs are not tested in our CI, we can't guarantee they actually works as expected.

Whenever we have someone willing to maintain the rootfs we can have them added back, and also add a CI job to test them altogether, avoiding then any possible regression.

This PR supersedes:
* #3325
* #3326 
* #3327 
* #3328

### CODE REVIEW

**Comment by fidencio** [comment]
Note, I've opened individual PRs before in order to give the chance of the distro maintainers to jump in to the issue.  We've got to an agreement there already, thus it was easier to create the this all-in-one PR rather than update one, have it merge, deal with conflicts in the next, and so on and so forth.

**Review by jodh-intel — APPROVED** [review]
Thanks @fidencio.

lgtm

**Comment by fidencio** [comment]
cloud-hypervisor failure:
```
10:45:57 not ok 2 Empty dir volume when FSGroup is specified with non-root container
10:45:59 # (from function `crictl_pull' in file tests_common.sh, line 34,
10:45:59 #  in test file k8s-empty-dirs.bats, line 51)
10:45:59 #   `crictl_pull "$image"' failed
10:45:59 # [bats-exec-test:23] INFO: k8s configured to use runtimeclass
10:45:59 # time="2022-01-11T09:45:58Z" level=fatal msg="pulling image: rpc error: code = FailedPrecondition desc = failed to pull and unpack image \"k8s.gcr.io/e2e-test-images/agnhost:2.21\": failed commit on ref \"layer-sha256:a7894ac8d81c85fa2694a0423e6dc256ea25bca5616cd56494d27b809c8cfd92\": \"layer-sha256:a7894ac8d81c85fa2694a0423e6dc256ea25bca5616cd56494d27b809c8cfd92\" failed size validation: 867177 != 10881087: failed precondition"
10:45:59 # Error from server (NotFound): pods "sharevol-kata" not found
10:45:59 # Error from server (NotFound): pods "sharevol-kata" not found
```

**Review by Jakob-Naucke — CHANGES_REQUESTED** [review]
@fidencio please check `grep -Er "(fedora|gentoo|suse)" tools/osbuilder/rootfs-builder`, there are some leftovers*

*the image base for Clear Linux is okay

**Comment by fidencio** [comment]
> @fidencio please check `grep -Er "(fedora|gentoo|suse)" tools/osbuilder/rootfs-builder`, there are some leftovers*

You're talking about the README, right?
```
rootfs-builder/README.md:   $ # build guest O/S rootfs based on fedora
rootfs-builder/README.md:   $ ./rootfs-builder/rootfs.sh -r "${PWD}/fedora_rootfs" fedora
rootfs-builder/README.md:   $ ./image-builder/image_builder.sh "${PWD}/fedora_rootfs"
rootfs-builder/README.md:  $ EXTRA_PKGS="vim emacs" ./rootfs-builder/rootfs.sh -r ${PWD}/myrootfs fedora
```

The rootfs ones seem to be related to the host OS, not to the guest.
```
rootfs-builder/README.md:   $ # build guest O/S rootfs based on fedora
rootfs-builder/README.md:   $ ./rootfs-builder/rootfs.sh -r "${PWD}/fedora_rootfs" fedora
rootfs-builder/README.md:   $ ./image-builder/image_builder.sh "${PWD}/fedora_rootfs"
rootfs-builder/README.md:  $ EXTRA_PKGS="vim emacs" ./rootfs-builder/rootfs.sh -r ${PWD}/myrootfs fedora
```

**Comment by Jakob-Naucke** [comment]
@fidencio ah yes, those in `rootfs.sh` are host distro. Sorry.

**Comment by fidencio** [comment]
> @fidencio ah yes, those in `rootfs.sh` are host distro. Sorry.
```
rootfs-builder/rootfs.sh:	gentoo)
rootfs-builder/rootfs.sh:		args+=" -v ${gentoo_local_portage_dir}:/usr/portage/packages"
rootfs-builder/rootfs.sh:		args+=" --volumes-from ${gentoo_portage_container}"
```
This bit is actually the guest bit, so I am removing this as well.

Thanks @Jakob-Naucke!

**Review by devimc — APPROVED** [review]
lgtm - bye bye Gentoo :disappointed:

**Review by Jakob-Naucke — APPROVED** [review]

File: tools/osbuilder/README.md [hunk 1]
```diff
@@ -80,8 +80,8 @@ filesystem components to generate an initrd.
 3. When generating an image, the initrd is extracted to obtain the base rootfs for
 the image.
 
-CentOS is the default distro for building the rootfs, to use a different one, you can set `DISTRO=<your_distro>`.
-For example `make USE_DOCKER=true DISTRO=ubuntu rootfs` will make Ubuntu rootfs using Docker.
+Ubuntu is the default distro for building the rootfs, to use a different one, you can set `DISTRO=alpine|clearlinux|debian|ubuntu`.
```
Comment (jcvenegas) [inline tools/osbuilder/README.md:83]: Speaking of defaults you will want to sync/update the values from https://github.com/kata-containers/kata-containers/blob/main/versions.yaml#L118-L132

**Comment by wainersm** [comment]
> lgtm - bye bye Gentoo disappointed

@devimc I've always been curious about Gentoo...was it introduced for "hobby" or because we had users?

**Review by wainersm — APPROVED** [review]

**Comment by wainersm** [comment]
@fidencio thanks for working on this!

IMHO the osbuilder should have its supported guest/host/build_method split among maintainers/owners and clear rules on when to add/remove support.

**Comment by devimc** [comment]
@wainersm 
 > @devimc I've always been curious about Gentoo...was it introduced for "hobby" or because we had users?

I was comparing all the distros looking for the best performing distro, surprisingly debian is the best

**Comment by wainersm** [comment]
> @wainersm
> 
> > @devimc I've always been curious about Gentoo...was it introduced for "hobby" or because we had users?
> 
> I was comparing all the distros looking for the best performing distro, surprisingly debian is the best

Cool!

**Comment by dgibson** [comment]
Good riddance.  This gives us a much smaller base to look at, which I hope will less us make osbuilder into something more robust and usable.

**Review by egernst — APPROVED** [review]
you had me at image

**Comment by Bevisy** [comment]
@fidencio Sorry to disturb, but our team has actually been using centos as the base image, even though we all know that centos is coming to an end. But I was wondering if we could keep centos in this cleanup for now, until we decide on a new distribution. I can continue to maintain the use of centos rootfs.

**Comment by Jakob-Naucke** [comment]
@Bevisy cf. #3341

**Comment by Bevisy** [comment]
> @Bevisy cf. #3341

Fine, also a good way.

File: tools/osbuilder/README.md (hunk 1 above)
Comment (Jakob-Naucke) [reply tools/osbuilder/README.md:83]: @fidencio https://github.com/kata-containers/kata-containers/pull/3328#issuecomment-999322121

Comment (fidencio) [reply tools/osbuilder/README.md:83]: Thanks, fixed!

**Pull request merged**